mdaviz/mda_file_table_model.py

```diff
@@ -0,0 +1,219 @@
+"""
+QAbstractTableModel of folder content.
+
+.. autosummary::
+
+    ~MDAFileTableModel
+"""
+from mda import readMDA
+from PyQt5 import QtCore
+import yaml
+from . import utils
+
+
+# For this purpose, p and d will all be called detectors
+# since you can plot d vs p, but also d vd d
+
+
+class MDAFileTableModel(QtCore.QAbstractTableModel):
+    checkboxToggled = QtCore.pyqtSignal(int, int)
+
+    def __init__(self, data, parent):
+        self.parent = parent
+        super().__init__()
+
+        self.actions_library = {
```

Comment:
I'm starting to believe this algorithm is not a great fit for the task at hand: picking data fields for the X, Y and MON columns of the plot.

There are selection rules to pick these fields.  The current internal data structure keeps track of the checkbox settings.  Instead, there should be an internal structure from which the proper checkbox settings are evaluated.  The proper settings (as part of the selection rules) could change other checkboxes.

1. A data field selection could have one of four states: unselected (`None`), `"X"`, `"Y"`, or `"Mon"`
2. Only zero or one data field can be selected as `"X"`
3. Only zero or one data field can be selected as `"Mon"`
4. One or more data fields can be selected as `"Y"`

The selection rules should accept the latest action by the user (checking or unchecking), apply reassignments to the other data fields per the above rules, then redisplay all affected checkboxes.

In addition to `setData()` and `flags()`, this code should have a `checkbox(self, index)` method that returns either `QtCore.Qt.Unchecked` or `QtCore.Qt.Checked` based on the internal structure.

A `setCheckbox(self, index, state)` receives the action from the user (via `setData()`), updates the internal structure (rule 1 above) and applies the remaining selection rules 2-4.